fiftyone/operators/operations.py

```diff
@@ -0,0 +1,323 @@
+"""
+FiftyOne operator execution.
+
+| Copyright 2017-2024, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import json
+
+from bson import json_util
+
+
+class Operations(object):
+    """Interface to trigger builtin operations on an execution context.
+
+    Args:
+        ctx: an :class:`fiftyone.operators.executor.ExecutionContext`
+    """
+
+    def __init__(self, ctx):
+        self._ctx = ctx
+
+    ###########################################################################
+    # Builtin Python operators
+    ###########################################################################
+
+    def clone_selected_samples(self):
+        """Clone the selected samples in the App."""
+        return self._ctx.trigger("clone_selected_samples")
+
+    def clone_sample_field(self, field_name, new_field_name):
+        """Clone a sample field to a new field name.
+
+        Args:
+            field_name: the name of the field to clone
+            new_field_name: the name for the new field
+        """
+        return self._ctx.trigger(
+            "clone_sample_field",
+            params={
+                "field_name": field_name,
+                "new_field_name": new_field_name,
+            },
+        )
+
+    def rename_sample_field(self, field_name, new_field_name):
+        """Rename a sample field to a new field name.
+
+        Args:
+            field_name: the name of the field to rename
+            new_field_name: the new name for the field
+        """
+        return self._ctx.trigger(
+            "rename_sample_field",
+            params={
+                "field_name": field_name,
```

Comment:
@ritch just checking: does `resolve_input()` get called when triggering operations via `ctx.trigger()`?

Asking because in the vast majority of existing operations, parameter validation is _only_ performed in `resolve_input()`, and `execute()` assumes that all provided data is valid. If this method of triggering operations bypasses validation, then all kinds of bad things could happen.

Reply:
`ctx.trigger` at the moment does not prompt user for input even if input is required for an operator. However, we do perform validation Python operator prior to execution. However, it seems like we do not do the same for JavaScript operator so we should add execution time validation for JavaScript operators. The failure due to validation should get logged in the console for Python operators. The follow-up improvement unrelated to the PR could be:

- Add JS execution time validation (if indeed missing)
- Show validation error (and other errors) in a palette even when invoked with trigger

Reply:
@imanjra thank you! I've created a separate Jira ticket for your proposed follow-up improvements 👍